hfile_s3.c

```diff
@@ -555,29 +557,73 @@ static s3_auth_data * setup_auth_data(const char *s3url, const char *mode,
         if ((v = getenv("AWS_DEFAULT_PROFILE")) != NULL) kputs(v, &profile);
         else if ((v = getenv("AWS_PROFILE")) != NULL) kputs(v, &profile);
         else kputs("default", &profile);
+
+        if ((v = getenv("HTS_S3_ADDRESS_STYLE")) != NULL) {
+            if (strcasecmp(v, "virtual") == 0) {
+                address_style = s3_virtual;
+            } else if (strcasecmp(v, "path") == 0) {
+                address_style = s3_path;
+            }
+        }
     }
 
     if (ad->id.l == 0) {
+        kstring_t url_style = KS_INITIALIZE;
         const char *v = getenv("AWS_SHARED_CREDENTIALS_FILE");
         parse_ini(v? v : "~/.aws/credentials", profile.s,
                   "aws_access_key_id", &ad->id,
                   "aws_secret_access_key", &ad->secret,
                   "aws_session_token", &ad->token,
-                  "region", &ad->region, NULL);
+                  "region", &ad->region,
+                  "addressing_style", &url_style, NULL);
```

Comment:
You or I should have put this `NULL` on the next line — so extra entries can be added more clearly — ages ago :smile: (and similarly in the other `parse_ini` call below).

```suggestion
                  "addressing_style", &url_style,
                  NULL);
```

Reply:
Adjusted (will squash on merge).